From e4a210808273ef81361a9afabad364f1618ee7bb Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Mon, 9 Aug 2021 21:53:10 +0800
Subject: [PATCH 10/11] LoongArch: fix libdir search paths for elf64loongarch

First of all, target tuple matching in ld/emulparams/elf64loongarch.sh
is broken, so fix it.

Secondly, the previous value for LIBPATH_SUFFIX suggests a two-level
libdir layout, like RISC-V, which is known to cause much breakage.
Remove the two-level one and keep only the "64" part, which gives us
"lib64".
---
 ld/emulparams/elf64loongarch.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ld/emulparams/elf64loongarch.sh b/ld/emulparams/elf64loongarch.sh
index a9d800ad107..9c87696a55c 100644
--- a/ld/emulparams/elf64loongarch.sh
+++ b/ld/emulparams/elf64loongarch.sh
@@ -2,10 +2,10 @@
 OUTPUT_FORMAT="elf64-loongarch"
 
 case "$target" in
-  loong64*-linux*)
+  loongarch64*-linux*)
     case "$EMULATION_NAME" in
       *64*)
-	LIBPATH_SUFFIX="64/lib64 64";;
+	LIBPATH_SUFFIX=64;;
     esac
     ;;
 esac
-- 
2.33.0

